Sources/PrimerSDK/Classes/Core/3DS/ThreeDSecureService.swift

```diff
@@ -0,0 +1,207 @@
+    
+    func initializeSDK(completion: @escaping (Result<Void, Error>) -> Void) {
+        do {
+            let configBuilder = ConfigurationBuilder()
+            try configBuilder.log(to: .debug)
+            try configBuilder.license(key: Primer.netceteraLicenseKey)
+            
+            let supportedSchemeIds: [String] = ["A999999999"]
+                         
+            let scheme = Scheme(name: "scheme_name")
+            scheme.ids = supportedSchemeIds
+            scheme.encryptionKeyValue = Certificates.cer1
+            scheme.rootCertificateValue = Certificates.cer3
+            scheme.logoImageName = "visa"
+            
+            try configBuilder.add(scheme)
+             
+            let configParameters = configBuilder.configParameters()
+            
+            try threeDS2Service.initialize(configParameters,
+                                           locale: nil,
+                                           uiCustomization: nil)
+            completion(.success(()))
+        } catch {
+            completion(.failure(error))
+        }
+    }
+    
+    func verifyWarnings(completion: @escaping (Result<Void, Error>) -> Void) {
+        var sdkWarnings: [Warning] = []
+        do {
+            sdkWarnings = try threeDS2Service.getWarnings()
+        } catch {
+            completion(.success(()))
+            return
+        }
+        
+        if !sdkWarnings.isEmpty {
+            var message = ""
+            for warning in sdkWarnings {
+                message += warning.getMessage()
+                message += "\n"
+            }
+            
+            #if DEBUG
+            completion(.success(()))
+            #else
+            let err = NSError(domain: "netcetera", code: 100, userInfo: [NSLocalizedDescriptionKey: message])
+            completion(.failure(err))
+            #endif
+        } else {
+            completion(.success(()))
+        }
+    }
+    
+    func netceteraAuth(paymentMethod: PaymentMethodToken, protocolVersion: ThreeDS.ProtocolVersion, completion: @escaping (Result<Transaction, Error>) -> Void) {
+        do {
+            var directoryServerId: String
+            
+            switch paymentMethod.paymentInstrumentData?.network?.lowercased() {
+            case "visa":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .visa())
+            case "mastercard":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .mastercard())
+            case "diners":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .diners())
+            case "jcb":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .jcb())
+            case "amex":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .amex())
+            case "union":
+                directoryServerId = ThreeDS.directoryServerIdFor(scheme: .union())
+            default:
+                directoryServerId = "A999999999"
```

Comment:
[Q]
Shouldn't we stop 3DSecure if this happens?

Reply:
Yeah of course, we're still testing though and this is the only test card & test scheme available on 3dsecure.io. I had to leave that so you can actually test with a card.